### ISSUE TITLE
*READ-DEFAULT-FLOAT-FORMAT* is ignored when reading floats

### ISSUE THREAD

**Issue by stassats** [drmeister/clasp#61]
`*READ-DEFAULT-FLOAT-FORMAT*` => SINGLE-FLOAT
(type-of 1.2) => DOUBLE-FLOAT

incidentally, it should also be consulted when printing, non-default format should use the exponent marker, i.e. 1.2d0 when it's double-float, and 1.9f0 if it's  SINGLE-FLOAT, and no marker if it's the same format.

**Comment by Shinmera** [comment]
Can confirm on b05884f.
```
> *read-default-float-format*

SINGLE-FLOAT
> (type-of 1.2)

DOUBLE-FLOAT
```

**Comment by drmeister** [comment]
Fixed in fae0bd3

**Comment by drmeister** [comment]
I think the current version fixes all of the problems that stassats brought up

**Comment by Shinmera** [comment]
Fixed on 9c24f9a45d0bd4bd4a80cdcac18b7cee012947fb
```
> (type-of 1.2)

SINGLE-FLOAT
> (type-of (realpart #c(1.0 1.0)))

SINGLE-FLOAT
> (setf *read-default-float-format* 'double-float)

DOUBLE-FLOAT
> (type-of 1.2)

DOUBLE-FLOAT
```
